src/Illuminate/Cache/MemoizedStore.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Illuminate\Cache;
+
+use Illuminate\Contracts\Cache\Store;
+
+class MemoizedStore implements Store
+{
+    /**
+     * The memoized cache values.
+     *
+     * @var array<string, mixed>
+     */
+    protected $cache = [];
+
+    /**
+     * Create a new memoized cache instance.
+     *
+     * @param  string  $name
+     * @param  \Illuminate\Cache\Repository  $repository
+     */
+    public function __construct(
+        protected $name,
+        protected $repository,
+    ) {
+        //
+    }
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param  string  $key
+     * @return mixed
+     */
+    public function get($key)
+    {
+        $prefixedKey = $this->prefix($key);
+
+        if (array_key_exists($prefixedKey, $this->cache)) {
+            return $this->cache[$prefixedKey];
+        }
+
+        return $this->cache[$prefixedKey] = $this->repository->get($key);
```

Comment:
Octane and long-running processes will benefit more from this, right?

Shouldn't we have some mechanism to limit the number of keys kept, to avoid the internal array growing indefinitely?

Reply:
Each memo driver is bound as a scoped instance that is refreshed between octane requests and between queued jobs.

The intention is not to cache between executions but to memoize within the single execution.